styles/Canonical/025b-latinisms-to-reconsider.yml

```
@@ -0,0 +1,10 @@
+# 025 - Latinisms that we strongly advise against
+
+extends: existence
+message: Instead of 'a.m.', 'AM', 'p.m.', and 'PM', use 24-hour time.
+link: https://docs.ubuntu.com/styleguide/en#latin-words-and-phrases
+level: suggestion
+nonword: true
+tokens:
+  - '(?<![a-zA-Z!-\/[-`~])(?:a\.m\.?|A\.?M\.?|[0-9]\sam)(?![\w\.])'
+  - '(?<![a-zA-Z!-\/[-`~])[Pp]\.?[Mm]\.?(?![\w\.])'
```

Existing AM case does not capture lowercase at end of sentence, or lowercase with no space.

Best practices with Vale is to always use word boundaries if we're using `nonword: true`.

I think your current implementation might be an overcomplication. Limiting this rule to the most obvious practical uses will be more useful than having risky edge cases. For example, "PM" is an acronym for "product manager". 

```suggestion
  - '\b[1|2]?\d\s?(?:[AaPp]\.?[Mm]\.?)(?!\w+\b|\.)'
```

This would simplify the rule to only cases with specific times. Capturing specific cases is a lot easier than trying to implement broad scope ignores with negative lookbehinds.



This also means that we could use the captured text in the message- specifically providing advice about what to convert to 24 hour time.

---

Also your PM case doesn't catch "5P.M." - and I have no idea why. Maybe it's [Vale Studio](https://studio.vale.sh/s/6913bf4478966cdfe8ab0bdef70321f4).